### DISCUSSION TITLE
Using nelua from lua

### DISCUSSION THREAD

**Discussion by Dich0tomy** [edubart/nelua-lang#266] [Q&A] [answered]
Hello.

I am looking at this project as a possible alternative for things like `teal`, `fennel` and such, as well as `LuaCATS` from the lua-language-server which is a well established idiom for typing in lua.

Is it possible, for example, to write a neovim plugin with nelua, then transpile it somehow, or compile it and then use that as a plugin? What limitations would I encounter?

**Comment by Andre-LA** [comment, marked as answer]
Nelua transpiles to C and then to native, it cannot be required by Lua.

The only way to use it is to make a binding using the Lua C-API, in this case, isn't any different from the traditional Lua libraries written in C.

If you're using LuaJIT, it has a ffi module which, then, yes, you could import nelua code if your nelua code is being exported as a standalone C library:

https://nelua.io/overview/#exporting-named-c-functions

**Reply by Dich0tomy** [reply]
That's more or less what I expected, but I also hoped there was already some machinery which would streamline the process, thanks!

**Reply by Andre-LA** [reply]
I think it would be possible considering the meta-programming capabilities, but someone would need to code that first 😆
